homeassistant/components/light/__init__.py

```diff
@@ -245,12 +244,17 @@ def async_handle_light_service(service):
             else:
                 yield from light.async_toggle(**params)
 
-            if light.should_poll:
-                update_coro = light.async_update_ha_state(True)
-                if hasattr(light, 'async_update'):
-                    update_tasks.append(hass.loop.create_task(update_coro))
-                else:
-                    yield from update_coro
+        update_tasks = []
+
+        for light in target_lights:
+            if not light.should_poll:
+                continue
+
+            update_coro = light.async_update_ha_state(True)
```

Comment:
We should wrap this call in `hass.loop.create_task` so that if we first called this on 2 async entities, and then have to wait for a sync entity, the async entities can still continue to update.